src/SharpYaml/Emitter.cs

```diff
@@ -87,6 +87,48 @@ public class Emitter : IEmitter
 
         private bool isOpenEnded;
 
+        private readonly MutableStringLookAheadBuffer buffer = new MutableStringLookAheadBuffer();
```

Comment:
Does it really need to be a class?

Reply:
Since I'm using it via the ILookAheadBuffer interface, having it as a struct would end up doing memory allocation whenever I make a call on it, which would be worse.